Packages/com.chisel.components/Chisel/Components/API.public/BrushMeshAssetFactory/Capsule/BrushFactory.Capsule.cs

```diff
@@ -16,153 +16,151 @@ namespace Chisel.Components
 {
     // TODO: rename
     public sealed partial class BrushMeshAssetFactory
-	{
-		// possible situations:
-		//	capsule with top AND bottom set to >0 height
-		//	capsule with top OR bottom set to 0 height
-		//	capsule with both top AND bottom set to 0 height
-		//	capsule with height equal to top and bottom height
-		public static bool GenerateCapsuleVertices(CSGCapsuleDefinition definition, ref Vector3[] vertices)
-		{
-			definition.Validate();
-			var haveTopHemisphere		= definition.haveRoundedTop;
-			var haveBottomHemisphere	= definition.haveRoundedBottom;
-			var haveMiddleCylinder		= definition.haveCylinder;
+    {
+        // possible situations:
+        //	capsule with top AND bottom set to >0 height
+        //	capsule with top OR bottom set to 0 height
+        //	capsule with both top AND bottom set to 0 height
+        //	capsule with height equal to top and bottom height
+        public static bool GenerateCapsuleVertices(ref CSGCapsuleDefinition definition, ref Vector3[] vertices)
+        {
```

Comment:
making the definition a ref was necessary since Validate would ensure the definition had enough surfaceAssets allocated. since the definition is a struct, it wouldn't actually change the original without it. 
Probably a good idea to revisit this, to see if there's a cleaner way to do this.